packages/nextjs/pages/blockexplorer.tsx

```diff
@@ -0,0 +1,322 @@
+              >
+                Value (ETH)
+              </th>
+            </tr>
+          </thead>
+          <tbody className="bg-base-100 divide-y divide-primary-content text-base-content">
+            {Array.from(blocks.reduce((map, block) => map.set(block.hash, block), new Map()).values()).map(block =>
+              block.transactions.map((tx: TransactionWithFunction) => {
+                const receipt = transactionReceipts[tx.hash];
+
+                const shortTxHash = `${tx.hash.substring(0, 6)}...${tx.hash.substring(tx.hash.length - 4)}`;
+                const date = new Date(block.timestamp * 1000);
+                const month = String(date.getMonth() + 1).padStart(2, "0");
+                const day = String(date.getDate()).padStart(2, "0");
+                const timeMined = `${month}/${day} ${date.toLocaleTimeString("en-US", {
+                  hour: "2-digit",
+                  minute: "2-digit",
+                  second: "2-digit",
+                  hour12: true,
+                })}`;
+                const functionCalled = tx.data.substring(0, 10);
+
+                return (
+                  <tr key={tx.hash} className="bg-base-200 hover:bg-base-300 transition-colors duration-200">
+                    <td className="border px-4 py-2 text-base-content">
+                      <Link className="text-base-content hover:text-accent-focus" href={`/transaction/${tx.hash}`}>
+                        {shortTxHash}
+                      </Link>
+                    </td>
+
+                    <td className="border px-4 py-2 w-full md:w-1/2 lg:w-1/5 text-base-content">
+                      {tx.functionName === "0x" ? "" : tx.functionName}
+                      {functionCalled !== "0x" && (
+                        <span className="ml-2 inline-block rounded-full px-3 py-1 text-sm font-semibold text-primary-content bg-accent">
+                          {functionCalled}
+                        </span>
+                      )}
+                    </td>
+                    <td className="border px-4 py-2 w-20 text-base-content">{block.number}</td>
+                    <td className="border px-4 py-2 text-base-content">{timeMined}</td>
+                    <td className="border px-4 py-2 text-base-content">
+                      <Address address={tx.from} />
+                    </td>
+                    <td className="border px-4 py-2 text-base-content">
+                      {!receipt?.contractAddress ? (
+                        tx.to && <Address address={tx.to} />
+                      ) : (
+                        <span>
+                          Contract Creation:
+                          <Address address={receipt.contractAddress} />
+                        </span>
+                      )}
+                    </td>
+                    <td className="border px-4 py-2 text-base-content">{ethers.utils.formatEther(tx.value)} ETH</td>
```

Comment:
native currency token instead of `ETH`

Reply:
I fixed this but as this block explorer will only be used with a local hardhat node, the currency will always be ETH. It might be better to keep it as text. This is something trivial so we might decide it later as well.